requirements.txt

```diff
@@ -1,15 +1,15 @@
-numpy
-accelerate>=0.20.3
-packaging
-transformers>=4.34.1
-torch
-aim==3.17.5
-sentencepiece
-tokenizers>=0.13.3
-tqdm
-trl
-ninja
-peft>=0.8.0
-datasets>=2.15.0
-fire
+numpy>=1.26.4,<2.0
+accelerate>=0.20.3,<1.0
+transformers>=4.34.1,<5.0
+torch>=2.2.0,<3.0
```

Comment:
I ran into an error while using transformers 4.38.2.I think we need to pin specific versions. For example `transformers==4.37.2`
```
Traceback (most recent call last):
  File "/Users/tedchang/pub_git/fms-hf-tuning/tuning/sft_trainer.py", line 312, in <module>
    fire.Fire(main)
  File "/Users/tedchang/.pyenv/versions/fms-hf-tuning/lib/python3.11/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tedchang/.pyenv/versions/fms-hf-tuning/lib/python3.11/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
                                ^^^^^^^^^^^^^^^^^^^^
  File "/Users/tedchang/.pyenv/versions/fms-hf-tuning/lib/python3.11/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tedchang/pub_git/fms-hf-tuning/tuning/sft_trainer.py", line 308, in main
    train(model_args, data_args, training_args, tune_config)
  File "/Users/tedchang/pub_git/fms-hf-tuning/tuning/sft_trainer.py", line 256, in train
    trainer = SFTTrainer(
              ^^^^^^^^^^^
  File "/Users/tedchang/.pyenv/versions/fms-hf-tuning/lib/python3.11/site-packages/trl/trainer/sft_trainer.py", line 295, in __init__
    super().__init__(
  File "/Users/tedchang/.pyenv/versions/fms-hf-tuning/lib/python3.11/site-packages/transformers/trainer.py", line 648, in __init__
    self.is_fsdp_xla_v2_enabled = args.fsdp_config["xla_fsdp_v2"]
                                  ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
KeyError: 'xla_fsdp_v2'
```

Reply:
Indeed, minor changes will still cause breaking changes and we need to discuss how to handle dependency upgrades and testing. This is meant to have some kind of upper bounds so as to not include further bigger changes but does not solve the dependency breaks completely

Reply:
These types of bugs can arrive anytime and in any version. I think the goal here should be:
1. Open enough range for pip dep version resolution
2. Min version of dep that is required to run the code
3. Max version that we think should not contain any breaking change
4. platform and other environment specifications 


@anhuong for above case, for transformers range, we can provide an exclusion as well, like: 

`transformers>=4.34.1,<5.0,!=4.38.2`

the other option would be to restrict the version of transformers to a specific version like in [this PR](https://github.com/foundation-model-stack/fms-hf-tuning/pull/68/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552R4)